### PR TITLE
Adds send tab functionality to UI

### PR DESCRIPTION
**Pull request by jonalmeida**
Updated build flag to only work on nightly/debug.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/share/ShareFragment.kt [hunk 1]
```diff
@@ -84,13 +117,13 @@ class ShareFragment : DialogFragment(), CoroutineScope {
...
                     startActivity(intent)
                 }
-                // TODO support other actions in a follow-up issue
             }
             dismiss()
         }
     }
 
     companion object {
+        // TODO Replace this link with the correct one when provided.
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/share/ShareFragment.kt]: I'm not really sure what this is supposed to do even. "Add device" was in the mocks, but perhaps we should remove it from the UI for now until it does something useful?

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/share/ShareFragment.kt]: Add device was removed from the mocks (in error), so the functionality was missing on what it was supposed to do. This was our backup solution, but some other offline conversations lead us to showing a prompt instead which I'm going to add to this PR.

**Comment by jonalmeida** [comment]
> and I don't see where you actually use the build flag?

This was added in the previous PR, we only changed what qualifies the enable/disable in this PR.

File: app/src/main/java/org/mozilla/fenix/share/ShareFragment.kt (hunk 1 above)
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/share/ShareFragment.kt]: I'll fix this in a follow-up.

**Pull request merged**